corehq/apps/sso/forms.py

```diff
@@ -642,6 +735,19 @@ def clean_entity_id(self):
         _ensure_entity_id_matches_expected_provider(entity_id, self.idp)
         return entity_id
 
+    def clean_max_days_until_user_api_key_expiration(self):
+        enforce_expiration = bool(self.data.get('enforce_user_api_key_expiration'))
+        if not enforce_expiration:
+            return None
+
+        return self.cleaned_data['max_days_until_user_api_key_expiration']
+
+    def clean(self):
+        if not self.has_error('max_days_until_user_api_key_expiration'):
+            enforces_expiration = bool(self.cleaned_data['enforce_user_api_key_expiration'])
+            if enforces_expiration and not self.cleaned_data.get('max_days_until_user_api_key_expiration', None):
+                self.add_error('max_days_until_user_api_key_expiration', gettext('a value must be specified'))
```

Comment:
how does this error appear to the user? As I mentioned in Jing's PR, we historically have thrown validation errors in the `clean_<field_slug>` method due to the superior positioning of the error next to the field with an issue. It's more clear to the user when this is the case. I believe when we use the `clean` method, the errors appear as a bulleted list at the top of the form, which is a less ideal experience.

Reply:
rescinding my comment about `clean` as I just saw your comment about `self.add_error`.

Reply:
```suggestion
                self.add_error('max_days_until_user_api_key_expiration', gettext('Please specify a value.'))
```

Reply:
Addressed in 8b43ece8ae4a67e73af1861bf8d9b53bc0982263